src/databricks/labs/remorph/coverage/coverage_utils.py

```diff
@@ -0,0 +1,125 @@
+    :return: Returns the environment variable's value, or None if it's not set and not required.
+    """
+    value = os.getenv(env_var)
+    if value is None and required:
+        message = f"Environment variable {env_var} is not set"
+        raise ValueError(message)
+    return value
+
+
+def get_current_commit_hash() -> str | None:
+    try:
+        return (
+            subprocess.check_output(
+                ["/usr/bin/git", "rev-parse", "--short", "HEAD"],
+                cwd=Path(__file__).resolve().parent,
+            )
+            .decode("ascii")
+            .strip()
+        )
+    except (subprocess.CalledProcessError, FileNotFoundError) as e:
+        logger.warning(f"Could not get the current commit hash. {e!s}")
+        return None
+
+
+def collect_transpilation_stats(
+    project: str,
+    commit_hash: str,
+    version: str,
+    source_dialect: type[Dialect],
+    target_dialect: type[Dialect],
+    input_dir: Path,
+    result_dir: Path,
+):
+    if not input_dir.exists() or not input_dir.is_dir():
+        message = f"The input path {input_dir} doesn't exist or is not a directory"
+        raise NotADirectoryError(message)
+
+    if not result_dir.exists():
+        logger.info(f"Creating the output directory {result_dir}")
+        result_dir.mkdir(parents=True)
+    elif not result_dir.is_dir():
+        message = f"The output path {result_dir} exists but is not a directory"
+        raise NotADirectoryError(message)
+
+    source_dialect_name = source_dialect.__name__
+    target_dialect_name = target_dialect.__name__
+    report_file_path = result_dir / f"{project}_{source_dialect_name}_{target_dialect_name}.jsonl".lower()
+
+    with open(report_file_path, "w", encoding="utf8") as report_file:
+        for input_file in get_supported_sql_files(input_dir):
+            sql = input_file.read_text(encoding="utf-8-sig")
+            report_entry = {
+                "project": project,
+                "commit_hash": commit_hash,
+                "version": version,
+                "source_dialect": source_dialect_name,
+                "target_dialect": target_dialect_name,
+                "timestamp": datetime.utcnow().isoformat(),
+                "filename": str(input_file.absolute().relative_to(input_dir.parent.absolute())),
+                "parsing_status": "",
+                "queries_parsed": 0,
+                "parsing_error": "",
+                "transpilation_status": "",
+                "queries_transpiled": 0,
+                "transpilation_error": "",
+            }
+            try:
+                expressions = [
+                    expression
+                    for expression in parse(
+                        sql,
+                        read=source_dialect,
+                        error_level=ErrorLevel.RAISE,
+                    )
+                    if expression
+                ]
+                report_entry["parsing_status"] = "successful"
```

Comment:
```suggestion
                report_entry["parsed"] = 1
```
we need to make it friendly with spark - we'll have to build a report like `ROUND(SUM(transpiled) / SUM(parsed)) AS success_rate` downstream